openfecli/parameters/plan_network_options.py

```diff
@@ -0,0 +1,123 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+"""Pydantic models for the definition of advanced CLI options
+
+"""
+import click
+from pydantic import BaseModel, ConfigDict
+from plugcli.params import Option
+from typing import Any, Optional
+import yaml
+import warnings
+
+
+class MapperSelection(BaseModel):
+    model_config = ConfigDict(extra='allow', str_to_lower=True)
+
+    method: str = 'LomapAtomMapper'
+    settings: dict[str, Any] = {}
+
+
+class NetworkSelection(BaseModel):
+    model_config = ConfigDict(extra='allow', str_to_lower=True)
+
+    method: str = 'generate_minimal_spanning_network'
+    settings: dict[str, Any] = {}
+
+
+class CliOptions(BaseModel):
+    model_config = ConfigDict(extra='allow')
+
+    mapper: Optional[MapperSelection] = None
+    network: Optional[NetworkSelection] = None
+
+
+def parse_yaml_planner_options(contents: str) -> CliOptions:
+    """Parse and minimally validate a user provided yaml
+
+    Parameters
+    ----------
+    contents : str
+      raw yaml formatted input to parse
+
+    Returns
+    -------
+    options : CliOptions
+      will have keys for mapper and network topology choices
+
+    Raises
+    ------
+    ValueError
+      for any malformed inputs
+    """
+    raw = yaml.safe_load(contents)
+
+    if False:
+        # todo: warnings about extra fields we don't expect?
+        expected = {'mapper', 'network'}
+        for field in raw:
+            if field in expected:
+                continue
+            warnings.warn(f"Ignoring unexpected section: '{field}'")
+
+    return CliOptions(**raw)
+
+
+def load_yaml_planner_options(path, *args, **kwargs) -> dict:
+    """Load cli options from yaml file path and resolve these"""
+    from openfe.setup.ligand_network_planning import (
+        generate_radial_network,
+        generate_minimal_spanning_network,
+        generate_maximal_network,
+        generate_minimal_redundant_network,
+    )
+    from openfe.setup import (
+        LomapAtomMapper,
+    )
+    from functools import partial
+
+    with open(path, 'r') as f:
+        raw = f.read()
+
+    opt = parse_yaml_planner_options(raw)
+
+    choices = {}
+
+    if opt.mapper:
+        mapper_choices = {
+            'lomap': LomapAtomMapper,
+            'lomapatommapper': LomapAtomMapper,
+        }
+
+        try:
+            cls = mapper_choices[opt.mapper.method]
+        except KeyError:
+            raise KeyError(f"Bad mapper choice: '{opt.mapper.method}'")
+
+        choices['mapper'] = cls(**opt.mapper.settings)
+    if opt.network:
+        network_choices = {
```

Comment:
@dwhswenson avert your eyes!  This is the hacky job I think will initially serve us.  I think having this dict-like thing being eventually much more clever is what you're envisaging, this is as far as I'm taking it.